packages/forklift-console-plugin/src/modules/Providers/ProviderRow.tsx

```diff
@@ -36,66 +39,70 @@ interface CellProps {
   currentNamespace?: string;
 }
 
-const ProviderPhaseToStatus = (phase: ProviderStatus): 'Error' | 'Ok' | 'Unknown' | 'Loading' => {
-  switch (phase) {
-    case 'Ready':
-      return 'Ok';
-    case 'ConnectionFailed':
-    case 'ValidationFailed':
-      return 'Error';
-    case 'Staging':
-      return 'Loading';
-    default:
-      return 'Unknown';
-  }
+const phaseToStatus = (phase: ProviderStatus): 'Error' | 'Ok' | 'Unknown' | 'Loading' => {
+  const phaseToStatusMap = {
+    Ready: 'Ok',
+    ConnectionFailed: 'Error',
+    ValidationFailed: 'Error',
+    Staging: 'Loading',
+  };
+
+  return phaseToStatusMap[phase] || 'Undefined';
+};
+
+const phaseToLabel = (phase: ProviderStatus, t): string => {
+  const phaseToLabelMap = {
+    Ready: t('Ready'),
+    ConnectionFailed: t('Connection Failed'),
+    ValidationFailed: t('Validation Failed'),
+    Staging: t('Staging'),
+  };
+
+  return phaseToLabelMap[phase] || t('Undefined');
 };
 
-const CategoryToIconMap = {
-  Critical: <RedExclamationCircleIcon />,
-  Error: <RedExclamationCircleIcon />,
-  Required: <GreenCheckCircleIcon />,
-  Warn: <BlueInfoCircleIcon />,
-  Advisory: <GreenCheckCircleIcon />,
+const ConditionToIcon = (condition: V1beta1ProviderStatusConditions): React.ReactNode => {
+  const categoryToIconMap = {
```

Comment:
The mapping is good enough as safety net - assuming that the backend is using only  conditions with `Status: True` (which seems the current approach)